### DISCUSSION TITLE
TLS has invalid file check

### DISCUSSION THREAD

**Discussion by cgm-aw** [belaban/JGroups#857] [Q&A]
Hi,

in JGroups 5.3.10.Final, TLS#init() has this check (line 123ff):
```
if(keystore_path != null) {
    File tmp = new File(keystore_path);
    if (!tmp.exists())
        throw new FileNotFoundException(keystore_path);
}
```

However, this check fails for relative paths. If I just store my keystore under src/main/resources/keystore.jks and I set keystore_path to "keystore.jks", the check will fail.

The code in `SslContextFactory.java#loadKeyStore()` handles the file loading correctly, but it does not fail if the key store is not found at all, it just creates a new key store then.

Since the file existence check only works for `keystore_path`, which is obsolete anyway, I'm not sure what the expected behavior is, so I did not create a PR.

My suggestion:
* Either alter `SslContextFactory.java#loadKeyStore()` to throw an exception if the InputStream for `KeyStore#load()` is null and remove the file check from `TLS#init()` - this might break existing implementations though, maybe you want to have the key store optional for some other use cases
* Or change the file check in `TLS#init()` to mimic the file handling of `SslContextFactory#loadKeyStore()` - this should then validate the field `truststore_path` and not `keystore_path`
* Or remove the file check altogether, since it validates a deprecated field :)

**Comment by belaban** [comment]
OK. I think a simple fix would be to check for an absolute path, then - if not found - a relative path. I think I even have code to do that in `Util`. I created [1] to keep track of this, hope to fix it this or next week.
[1] https://issues.redhat.com/browse/JGRP-2846

**Comment by belaban** [comment]
`SYM_ENCRYPT.init()` also does the right thing; I'll therefore either copy the code from that class or `loadKeystore()` over to `TLS`.

**Comment by belaban** [comment]
I removed the check for the existence of keystore_path
